JetNews/app/src/main/java/com/example/jetnews/ui/home/HomeRoute.kt

```diff
@@ -0,0 +1,182 @@
+ * @param openDrawer (event) request opening the app drawer
+ * @param scaffoldState (state) state for the [Scaffold] component on this screen
+ */
+@Composable
+fun HomeRoute(
+    homeViewModel: HomeViewModel,
+    openDrawer: () -> Unit,
+    scaffoldState: ScaffoldState = rememberScaffoldState()
+) {
+    // UiState of the HomeScreen
+    val uiState by homeViewModel.uiState.collectAsState()
+
+    HomeRoute(
+        uiState = uiState,
+        onToggleFavorite = { homeViewModel.toggleFavourite(it) },
+        onSelectPost = { homeViewModel.selectArticle(it) },
+        onRefreshPosts = { homeViewModel.refreshPosts() },
+        onErrorDismiss = { homeViewModel.errorShown(it) },
+        onInteractWithList = { homeViewModel.interactedWithList() },
+        onInteractWithDetail = { homeViewModel.interactedWithDetail(it) },
+        openDrawer = openDrawer,
+        scaffoldState = scaffoldState,
+    )
+}
+
+/**
+ * Displays the Home route.
+ *
+ * This composable is not coupled to any specific state management.
+ *
+ * @param uiState (state) the data to show on the screen
+ * @param onToggleFavorite (event) toggles favorite for a post
+ * @param onRefreshPosts (event) request a refresh of posts
+ * @param onErrorDismiss (event) error message was shown
+ * @param openDrawer (event) request opening the app drawer
+ * @param scaffoldState (state) state for the [Scaffold] component on this screen
+ */
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+fun HomeRoute(
+    uiState: HomeUiState,
+    onToggleFavorite: (String) -> Unit,
+    onSelectPost: (String) -> Unit,
+    onRefreshPosts: () -> Unit,
+    onErrorDismiss: (Long) -> Unit,
+    onInteractWithList: () -> Unit,
+    onInteractWithDetail: (String) -> Unit,
+    openDrawer: () -> Unit,
+    scaffoldState: ScaffoldState
+) {
+    // Construct the lazy list states for the list and the details outside of deciding which one to
+    // show.
+    // This allows the associated state to survive beyond that decision, and therefore we get to
+    // preserve the scroll throughout any changes to the content.
+    val homeListLazyListState = rememberLazyListState()
+    val articleDetailLazyListStates = when (uiState) {
+        is HomeUiState.HasPosts -> uiState.postsFeed.allPosts
+        is HomeUiState.NoPosts -> emptyList()
+    }.associate { post ->
+        key(post.id) {
+            post.id to rememberLazyListState()
+        }
+    }
+
+    BoxWithConstraints {
+        val useListDetail = when (getWindowSize(maxWidth)) {
+            WindowSize.Compact, WindowSize.Medium -> false
+            WindowSize.Expanded -> true
+        }
+
+        // Determine which type of the home screen to display
+        val homeScreenType = if (useListDetail) {
+            HomeScreenType.ListArticle
+        } else {
+            when (uiState) {
+                is HomeUiState.HasPosts -> if (uiState.isArticleOpen) {
```

Comment:
Formatting here is weird and makes reading the logic a bit harder. Can we move the if statement to a new line?